src/main/java/org/truffleruby/Layouts.java

```diff
@@ -101,6 +101,14 @@
 
 public abstract class Layouts {
 
+    // Standard identifiers
+    // These must appear before the generated layout list so the identifiers have been initialized by the time
+    // the layout singletons are created.
+
+    public static final HiddenKey OBJECT_ID_IDENTIFIER = new HiddenKey("object_id");
```

Comment:
We tracked this down to `Thread.guarding?`. While we can probably improve that method, is there an issue with adding the property here? We have space in the object and it'll avoid a second shape should `object_id` be called.